nectar/src/bitcoin/bitcoind.rs

```diff
@@ -188,6 +189,98 @@ impl Client {
         Ok(txid)
     }
 
+    pub async fn fund_htlc(
+        &self,
+        wallet_name: &str,
+        address: Address,
+        amount: Amount,
+    ) -> anyhow::Result<OutPoint> {
+        let address = address.to_string();
+
+        let response: CreatePsbtResponse = self
+            .rpc_client
+            .send_with_path(
+                format!("/wallet/{}", wallet_name),
+                jsonrpc::Request::new(
+                    "walletcreatefundedpsbt",
```

Comment:
Remind me of thor code... time to have a dedicated bitcoin rpc lib 🙃

Reply:
Once I have my lib ready, I was thinking of proposing to the `rust-bitcoin` lib to add the RPC stuff to it under a feature flag. Given that my lib is client-agnostic, the only dependency that would be added is `syn` but also that would be feature-flagged.

That would allow us/them to keep the types in the interface up to date and it would avoid situations in which we can no longer update rust-bitcoin because it would become incompatible with their rpc lib.

Reply:
Unless we are just better at maintaining our bitcoin rpc lib, there isn't really an advantage to it as long as it is two different libraries.